vnext/Microsoft.ReactNative.Cxx/JSValue.cpp

```diff
@@ -439,13 +439,13 @@ void JSValueArray::WriteTo(IJSValueWriter const &writer) const noexcept {
 JSValue::JSValue(JSValue &&other) noexcept : m_type{other.m_type} {
   switch (m_type) {
     case JSValueType::Object:
-      new (&m_object) JSValueObject(std::move(other.m_object));
+      new (std::addressof(m_object)) JSValueObject(std::move(other.m_object));
```

Comment:
any reason to use placement new + addressof instead of something like `m_object = std::move(other.m_object)` (same elsewhere)

Reply:
These are not real fields - these are part of a union. So, in constructor they are just like some memory initialized with 0. Doing an assignment may crash because we did not run default constructors there yet, and running destructors as a part of the assignment is not valid.